Sources/NIOPosix/MultiThreadedEventLoopGroup.swift

```diff
@@ -376,7 +376,11 @@ extension ScheduledTask: CustomStringConvertible {
 extension ScheduledTask: Comparable {
     @usableFromInline
     static func < (lhs: ScheduledTask, rhs: ScheduledTask) -> Bool {
-        return lhs._readyTime < rhs._readyTime
+        if lhs._readyTime == rhs._readyTime {
```

Comment:
Please add a comment to `id` in line 343, explaining that the `id` must be unique within an EventLoop and increasing for new scheduled tasks. The reason for this being that we use the id and the readyTime to determine ordering.